newm-server/src/main/kotlin/io/newm/server/features/user/verify/SpotifyProfileUrlVerifier.kt

```diff
@@ -0,0 +1,94 @@
+package io.newm.server.features.user.verify
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.auth.providers.BearerTokens
+import io.ktor.client.plugins.auth.providers.bearer
+import io.ktor.client.request.accept
+import io.ktor.client.request.basicAuth
+import io.ktor.client.request.forms.submitForm
+import io.ktor.client.request.get
+import io.ktor.http.ContentType
+import io.ktor.http.parameters
+import io.ktor.server.application.ApplicationEnvironment
+import io.ktor.util.logging.Logger
+import io.newm.server.features.user.model.TokenInfo
+import io.newm.server.ktx.getSecureConfigString
+import io.newm.shared.koin.inject
+import io.newm.shared.ktx.getConfigString
+import io.newm.shared.ktx.info
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.koin.core.parameter.parametersOf
+
+class SpotifyProfileUrlVerifier(
+    private val applicationEnvironment: ApplicationEnvironment,
+    private val httpClient: HttpClient,
+) : OutletProfileUrlVerifier {
+    private val logger: Logger by inject { parametersOf(javaClass.simpleName) }
+
+    private var bearerTokens: BearerTokens? = null
+    private lateinit var authorizedHttpClient: HttpClient
+
+    override suspend fun verify(outletProfileUrl: String, stageOrFullName: String) {
+        authorizeClient()
+        val spotifyProfileId = outletProfileUrl.substringAfterLast("/")
+        val response: SpotifyArtistResponse = authorizedHttpClient.get(
+            "https://api.spotify.com/v1/artists/$spotifyProfileId"
+        ) {
+            accept(ContentType.Application.Json)
+        }.body()
+        logger.info { "Spotify profile response for $spotifyProfileId : $response" }
+        if (response.name != stageOrFullName) {
+            throw IllegalArgumentException("Spotify profile name (${response.name}) does not match stageOrFullName ($stageOrFullName) for $spotifyProfileId")
+        }
+    }
+
+    private suspend fun authorizeClient() {
+        if (!::authorizedHttpClient.isInitialized) {
+            authorizedHttpClient =
```

Comment:
Maybe this `authorizedHttpClient` can be initialized in the declaration using `by lazy { }`

Reply:
It cannot. You cannot call `suspend` functions inside of `by lazy{ }` like we're doing with `refreshBearerTokens()`.